Core/src/org/sleuthkit/autopsy/contentviewers/MessageContentViewer.java

```diff
@@ -498,10 +502,54 @@ private static boolean isMessageArtifact(BlackboardArtifact nodeArtifact) {
                 || artifactTypeID == TSK_MESSAGE.getTypeID();
     }
 
+    /**
+     * Returns the artifact represented by node.
+     *
+     * If the node lookup has an artifact, that artifact is returned. However,
+     * if the node lookup is a file, then we look for a TSK_ASSOCIATED_OBJECT
+     * artifact on the file, and if a message artifact is found associated with
+     * the file, that artifact is returned.
+     *
+     * @param node Node to check.
+     * @return Blackboard artifact for the node, null if there isn't any.
+     */
+    private BlackboardArtifact getNodeArtifact(Node node) {
+        BlackboardArtifact nodeArtifact = node.getLookup().lookup(BlackboardArtifact.class);
+
+        if (nodeArtifact == null) {
+            try {
+                SleuthkitCase tskCase = Case.getCurrentCaseThrows().getSleuthkitCase();
+                AbstractFile file = node.getLookup().lookup(AbstractFile.class);
```

Comment:
Is it guaranteed that we will have an  AbstractFile in Node?

Reply:
No, there's no guarantee..  I did have a null check that got removed in the shuffle.  I'll update it.